fairscale/utils/parallel.py

```diff
@@ -87,15 +91,21 @@ def get_process_group_cached(ranks: Optional[List[int]] = None) -> ProcessGroup:
     if not dist.is_initialized():
         raise RuntimeError("torch.distributed is not yet initialized but process group is requested.")
 
+    # Init the cache if needed.
     if not hasattr(get_process_group_cached, "_global_group_cache"):
         get_process_group_cached._global_group_cache = {}  # type: ignore
-    cache = get_process_group_cached._global_group_cache  # type: ignore
+        # Populate with default process group.
+        cache = get_process_group_cached._global_group_cache  # type: ignore
+        assert dist.group.WORLD is not None
+        cache[None] = dist.group.WORLD
+        cache[list(range(dist.get_world_size()))] = dist.group.WORLD
 
-    if ranks is None:
-        ranks = list(range(dist.get_world_size()))
-
-    ranks_set = frozenset(ranks)  # take care of ordering and duplicates in the ranks list.
-    if ranks_set not in cache:
-        cache[ranks_set] = dist.new_group(list(ranks_set))
+    # Lookup and fill the cache if needed.
+    cache = get_process_group_cached._global_group_cache  # type: ignore
+    if ranks is not None:
+        # take care of ordering and duplicates in the ranks list.
+        ranks = sorted(list(set(ranks)))
+    if ranks not in cache:
+        cache[ranks] = dist.new_group(ranks)
```

Comment:
Given lists are unhashable how can we use it as a key here? I think `sorted` returns a list right?

Reply:
great catch. The test also caught it. just fixed it.

Reply:
push a better fix.